google/cloud/bigtable/examples/data_snippets.cc

```diff
@@ -687,6 +677,233 @@ void RowExists(google::cloud::bigtable::Table table, int argc, char* argv[]) {
+                  char* argv[]) {
+  if (argc != 5) {
+    throw Usage{
+        "rename-column <project-id> <instance-id>"
+        " <table-id> key family old-name new-name"};
+  }
+
+  auto key = ConsumeArg(argc, argv);
+  auto family = ConsumeArg(argc, argv);
+  auto old_name = ConsumeArg(argc, argv);
+  auto new_name = ConsumeArg(argc, argv);
+
+  // [START bigtable_mutate_mix_match]
+  namespace cbt = google::cloud::bigtable;
+  using google::cloud::Status;
+  using google::cloud::StatusOr;
+  [](cbt::Table table, std::string key, std::string family,
+     std::string old_name, std::string new_name) {
+    StatusOr<std::pair<bool, cbt::Row>> row =
+        table.ReadRow(key, cbt::Filter::ColumnName(family, old_name));
+
+    if (!row) {
+      throw std::runtime_error(row.status().message());
+    }
+    if (!row->first) {
+      throw std::runtime_error("Cannot find row " + key);
+    }
+
+    cbt::SingleRowMutation mutation(key);
+    for (auto const& cell : row->second.cells()) {
+      // Create a new cell
+      auto timestamp_in_milliseconds =
+          std::chrono::duration_cast<std::chrono::milliseconds>(
+              cell.timestamp());
+      mutation.emplace_back(cbt::SetCell(
+          family, new_name, timestamp_in_milliseconds, cell.value()));
+    }
+    mutation.emplace_back(cbt::DeleteFromColumn("fam", old_name));
+
+    google::cloud::Status status = table.Apply(std::move(mutation));
+    if (!status.ok()) {
+      throw std::runtime_error(status.message());
+    }
+    std::cout << "Row successfully updated\n";
+  }
+  // [END bigtable_mutate_mix_match]
+  (std::move(table), key, family, old_name, new_name);
+}
+
+void InsertTestData(google::cloud::bigtable::Table table, int argc,
+                    char* argv[]) {
+  if (argc != 1) {
+    throw Usage{"insert-test-data <project-id> <instance-id> <table-id>"};
+  }
+
+  // Write several rows in a single operation, each row has some trivial data.
+  // This is not a code sample in the normal sense, we do not display this code
+  // in the documentation. We use it to populate data in the table used to run
+  // the actual examples during the CI builds.
+  google::cloud::bigtable::BulkMutation bulk;
+  for (int i = 0; i != 5000; ++i) {
+    // Note: This example uses sequential numeric IDs for simplicity, but
+    // this can result in poor performance in a production application.
+    // Since rows are stored in sorted order by key, sequential keys can
+    // result in poor distribution of operations across nodes.
+    //
+    // For more information about how to design a Bigtable schema for the
+    // best performance, see the documentation:
+    //
+    //     https://cloud.google.com/bigtable/docs/schema-design
+    char buf[32];
+    snprintf(buf, sizeof(buf), "key-%06d", i);
+    google::cloud::bigtable::SingleRowMutation mutation(buf);
+    mutation.emplace_back(google::cloud::bigtable::SetCell(
+        "fam", "col0", "value0-" + std::to_string(i)));
```

Comment:
I couldn't tell whether skipping "value1" was intentional or not.

Reply:
There may have been a reason to skip `col1`, and I bet it was clever. Fixed.